scripts/build.sh

```diff
@@ -0,0 +1,293 @@
+IMAGE_VERSION=$(git describe --always --dirty --tags)
+IMAGE_TAG=$(echo $IMAGE_VERSION | cut -d. -f1,2)
+IMAGE_BRANCH=$(git rev-parse --abbrev-ref HEAD | sed 's/\///g')
+GIT_REF_SHORT=$(git rev-parse --short=8 --verify HEAD)
+GIT_REF_LONG=$(git rev-parse --verify HEAD)
+
+BUILDMNT=/go/src/$GOTARGET
+BUILD_IMAGE=golang:1.16
+AMD_IMAGE=gcr.io/distroless/static:nonroot
+ARM_IMAGE=gcr.io/distroless/static:nonroot-arm64
+WIN_IMAGE=mcr.microsoft.com/windows/servercore:1809
+TEST_IMAGE="sonobuoy/testimage:v0.1"
+
+unit_local() {
+	GODEBUG=x509ignoreCN=0 go test ${VERBOSE:+-v} -timeout 60s -coverprofile=coverage.txt -covermode=atomic $GOTARGET/cmd/... $GOTARGET/pkg/...
+}
+
+unit() {
+	docker run --rm -v $(pwd):$BUILDMNT -w $BUILDMNT $BUILD_IMAGE /bin/sh -c \
+    "GODEBUG=x509ignoreCN=0 go test ${VERBOSE:+-v} -timeout 60s -coverprofile=coverage.txt -covermode=atomic $GOTARGET/cmd/... $GOTARGET/pkg/..."
+}
+
+stress() {
+	docker run --rm -v $(pwd):$BUILDMNT -w $BUILDMNT $BUILD_IMAGE /bin/sh -c \
+    "GODEBUG=x509ignoreCN=0 go test ${VERBOSE:+-v} -timeout 60s -coverprofile=coverage.txt -covermode=atomic $GOTARGET/test/stress/..."
+}
+
+integration() {
+	docker run --rm \
+        -v $(pwd):$BUILDMNT \
+        -v /tmp/artifacts:/tmp/artifacts \
+        -v ${HOME}/.kube/config:/root/.kube/kubeconfig \
+        --env KUBECONFIG=/root/.kube/kubeconfig \
+        -w $BUILDMNT \
+        --env ARTIFACTS_DIR=/tmp/artifacts \
+        --env SONOBUOY_CLI=$SONOBUOY_CLI \
+        --env GODEBUG=x509ignoreCN=0 \
+        --network host \
+        $BUILD_IMAGE \
+    go test ${VERBOSE:+-v} -timeout 3m -tags=integration $GOTARGET/test/integration/...
+}
+
+lint() {
+	docker run --rm -v $(pwd):$BUILDMNT -w $BUILDMNT $BUILD_IMAGE /bin/sh -c \
+    "golint -set_exit_status ${VERBOSE:+-v} -timeout 60s $GOTARGET/cmd/... $GOTARGET/pkg/..."
+}
+
+vet() {
+	docker run --rm -v $(pwd):$BUILDMNT -w $BUILDMNT $BUILD_IMAGE /bin/sh -c \
+    "CGO_ENABLED=0 go vet ${VERBOSE:+-v} -timeout 60s $GOTARGET/cmd/... $GOTARGET/pkg/..."
+}
+
+pre() {
+	wget https://github.com/estesp/manifest-tool/releases/download/v1.0.1/manifest-tool-linux-amd64 \
+        -O manifest-tool
+	chmod +x ./manifest-tool
+	echo $DOCKERHUB_TOKEN | docker login --username sonobuoybot --password-stdin
+}
+
+# TODO: Make it easy to build single container for a specific arch
+build_container_dockerfile_image() {
+	docker build \
+        -t $REGISTRY/$2:$IMAGE_VERSION \
+        -t $REGISTRY/$2:$IMAGE_TAG \
+        -t $REGISTRY/$2:$IMAGE_BRANCH \
+        -t $REGISTRY/$2:$GIT_REF_SHORT \
+        -f $1 \
+		.
+}
+
+linux_containers() {
+	for arch in $LINUX_ARCH; do
+		if [ "$arch"="amd64" ]; then
+			sed -e "s|BASEIMAGE|$AMD_IMAGE|g" \
+			    -e 's|CMD1||g' \
+			    -e 's|BINARY|build/linux/amd64/sonobuoy|g' Dockerfile > Dockerfile-$arch
+			build_container_dockerfile_image Dockerfile-$arch sonobuoy
+			build_container_dockerfile_image Dockerfile-$arch sonobuoy-$arch
```

Comment:
do you think it's worth extracting this to a separate command? allowing people to just recreate the Dockerfile without building the container?

Reply:
Yea, that works with the other comment above about wanting to build just a single os/arch too. I'll split it into finer parts and compose as needed.